src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Report;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy strategy;
+
+    public ShopServiceImpl(OperationStrategy strategy) {
+        this.strategy = strategy;
+    }
+
+    @Override
+    public Report process(List<FruitTransaction> transactions) {
+        Map<String, Integer> fruitQuantityMap = new HashMap<>();
+        boolean isInitialization = true;
+        for (FruitTransaction transaction : transactions) {
+            isInitialization = validateAndUpdateState(isInitialization,
+                    transaction.getOperation());
+            OperationHandler handler = strategy.getHandler(transaction.getOperation());
+            handler.process(fruitQuantityMap, transaction);
+        }
+        return new Report(fruitQuantityMap);
```

Comment:
The service implementation should be independent and called in the main() method. Ensure that the `ShopServiceImpl` does not call methods from the `OperationStrategy` directly within its methods. Instead, it should be structured in a way that the `main()` method orchestrates the flow of the program by calling these services in the correct order.